### PR TITLE
[sdk/python] remove deprecation warning on result types unless using getters.

### LINKED ISSUE #11739: [codegen/python] deprecated warning for aws.secretsmanager.get_secret
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Calling `aws.secretsmanager.get_secret(...)` results in deprecated warnings for attributes of `GetSecretResult` even if these attributes are not accessed:

```
warning: rotation_enabled is deprecated: Use the aws_secretsmanager_secret_rotation data source instead
warning: rotation_lambda_arn is deprecated: Use the aws_secretsmanager_secret_rotation data source instead
warning: rotation_rules is deprecated: Use the aws_secretsmanager_secret_rotation data source instead
``` 

pulumi: 3.18.1
pulumi-aws: 4.29.0

### Steps to reproduce

1. Create a project with:
```python
import pulumi
import pulumi_aws as aws
secret_arn = aws.secretsmanager.get_secret(name="...").arn
```
2. Run `pulumi up`
3. Observe warnings:
```
warning: rotation_enabled is deprecated: Use the aws_secretsmanager_secret_rotation data source instead
warning: rotation_lambda_arn is deprecated: Use the aws_secretsmanager_secret_rotation data source instead
warning: rotation_rules is deprecated: Use the aws_secretsmanager_secret_rotation data source instead
```

### PR DESCRIPTION
**Pull request by dixler**
# Description

remove deprecation warning on result types unless using getters.

Fixes #11739 

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/codegen/testing/test/testdata/regress-py-12546/python/pulumi_plant/tree/v1/rubber_tree.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def __init__(__self__, *,
     @property
     @pulumi.getter
     def diameter(self) -> pulumi.Input['Diameter']:
+        warnings.warn("""Dear future maintainer, if there are changes here, make sure that this is printed before the value is set to the default or else this will always print.""", DeprecationWarning)
```
Comment (Frassle) [inline pkg/codegen/testing/test/testdata/regress-py-12546/python/pulumi_plant/tree/v1/rubber_tree.py:61]: Set to the default? Is this because it uses the same deprecation message as the input?

File: pkg/codegen/python/gen.go [hunk 2]
```diff
@@ -1809,9 +1804,10 @@ func (mod *modContext) genFunction(fun *schema.Function) (string, error) {
 			if i > 0 {
 				fmt.Fprintf(w, ",")
 			}
-			// Use the get_dict_value utility instead of calling __ret__.get directly in case the __ret__
-			// object has a get property that masks the underlying dict subclass's get method.
-			fmt.Fprintf(w, "\n        %[1]s=__ret__.%[1]s", PyName(ret.Name))
+			// Use the `pulumi.get()` utility instead of calling `__ret__.get` directly in case the `__ret__`
+			// object defines a get property with behavior other than simply returning the property value.
+			// `pulumi.get()` can only be used with classes decorated with `@input_type` or `@output_type`
+			fmt.Fprintf(w, "\n        %[1]s=pulumi.get(__ret__, '%[1]s')", PyName(ret.Name))
```
Comment (Frassle) [inline pkg/codegen/python/gen.go:1810]: Why has this changed to `pulumi.get`? Seems unrelated to the deprecated change.

File: pkg/codegen/testing/test/testdata/regress-py-12546/python/pulumi_plant/tree/v1/rubber_tree.py (hunk 1 above)
Comment (dixler) [reply pkg/codegen/testing/test/testdata/regress-py-12546/python/pulumi_plant/tree/v1/rubber_tree.py:61]: Correct. This was added to track a regression and dutifully spooked me on first read too (even though I added it).

File: pkg/codegen/python/gen.go (hunk 2 above)
Comment (dixler) [reply pkg/codegen/python/gen.go:1810]: It's based on the suggestion in the initial issue.

This sidesteps the `getter` function which causes instantiation to print deprecation messages that we're trying to hide from the user.

I'll update the comment to be more clear. :+1:

File: pkg/codegen/python/gen.go [hunk 3]
```diff
@@ -1809,9 +1804,10 @@ func (mod *modContext) genFunction(fun *schema.Function) (string, error) {
 			if i > 0 {
 				fmt.Fprintf(w, ",")
 			}
-			// Use the get_dict_value utility instead of calling __ret__.get directly in case the __ret__
-			// object has a get property that masks the underlying dict subclass's get method.
-			fmt.Fprintf(w, "\n        %[1]s=__ret__.%[1]s", PyName(ret.Name))
+			// Use the `pulumi.get()` utility instead of calling `__ret__.field` directly.
+			// This avoids the calling getter functions which will print deprecation messages on unused
```
Comment (Frassle) [inline pkg/codegen/python/gen.go:1808]:
```suggestion
			// This avoids calling getter functions which will print deprecation messages on unused
```

**Review by Frassle — APPROVED** [review]
Small typo otherwise seems reasonable

**Comment by dixler** [comment]
bors merge

**Comment by abhinav** [comment]
bors merge

**Comment by abhinav** [comment]
bors merge
